### PR TITLE
UI: Add CircleCI job

### PR DESCRIPTION
**Pull request by backspace**
This adds a job to test the UI on CircleCI, including the sort of branch
pattern-matching from #5839, so `.-ui/` branches only have that job
and not the non-UI ones.

I considered having an entire workflow for UI, which could have separate
jobs for linting vs Ember tests, but the lint commands take so little time
that it didn’t seem worth it.

There’s no use of nvm to change the Node version as the Docker image
is what controls that. It’s annoying to have to update the version in multiple
places, but probably infrequent.

### CODE REVIEW

**Comment by backspace** [comment]
Just to confirm that the branch filtering works, I pushed an empty commit on a branch I made off this one called `f-non-ui-circle-test` and you can see that [it has all the expected jobs](https://circleci.com/workflow-run/a0c68023-fe0f-4e13-ab78-f2f1d413ff31).

File: .circleci/config.yml [hunk 1]
```diff
@@ -153,6 +169,35 @@ jobs:
           path: /tmp/test-reports
       - store_artifacts:
           path: /tmp/test-reports
+  test-ui:
+    docker:
+      - image: circleci/node:10-browsers
+        environment:
+          JOBS: 2
```
Comment (notnoop) [inline .circleci/config.yml]: What is this variable for?  comment?

File: .circleci/config.yml [hunk 2]
```diff
@@ -5,38 +5,54 @@ references:
...
+    filters:
+      branches:
+        ignore: /^.-ui\b.*/
+
 
 workflows:
   build-test:
     jobs:
-      - lint
+      - lint:
```
Comment (endocrimes) [inline .circleci/config.yml]: we can rename this to `lint-go` to avoid confusion maybe?

File: .circleci/config.yml [hunk 3]
```diff
@@ -153,6 +169,35 @@ jobs:
...
+  test-ui:
+    docker:
+      - image: circleci/node:10-browsers
+        environment:
+          JOBS: 2
+    steps:
+      - checkout
+      - restore_cache:
+          keys:
+            - v1-deps-{{ .Branch }}-{{ checksum "ui/yarn.lock" }}
```
Comment (notnoop) [inline .circleci/config.yml]: I'd drop `branch` checks and only use checksum - allows us to reuse cache across branches effectively considering we rarely change yarn.lock and have a lot of short lived branches.

```suggestion
            - v1-deps-{{ checksum "ui/yarn.lock" }}
```

the same apply for save.

File: .circleci/config.yml (hunk 2 above)
Comment (backspace) [reply .circleci/config.yml]: 👍 will do, thanks!

File: .circleci/config.yml [hunk 4]
```diff
@@ -153,6 +169,35 @@ jobs:
...
+            - v1-deps-{{ .Branch }}-
+            - v1-deps-
+      - run:
+          name: yarn install
+          command: cd ui && yarn install
+      - save_cache:
+          key: v1-deps-{{ .Branch }}-{{ checksum "ui/yarn.lock" }}
+          paths:
+            - ./ui/node_modules
+      - run: export PATH="$HOME/.yarn/bin:$PATH"
```
Comment (notnoop) [inline .circleci/config.yml]: Unlike Travis - commands in circleci are executed on separate shells.  The export here has no effect for subsequent steps.  You can either export everytime it's needed (preferred) or use `BASH_ENV` hack, e.g. https://discuss.circleci.com/t/how-to-add-a-path-to-path-in-circle-2-0/11554/4 .

File: .circleci/config.yml (hunk 1 above)
Comment (backspace) [reply .circleci/config.yml]: Done, thanks.

File: .circleci/config.yml (hunk 3 above)
Comment (backspace) [reply .circleci/config.yml]: I was shamelessly copying [this](https://kevin.pfefferle.co/2018/01/27/circleci-2-with-ember) but yeah, agreed!

File: .circleci/config.yml (hunk 4 above)
Comment (backspace) [reply .circleci/config.yml]: heh nice, well this was doing nothing then! I removed it.

**Pull request merged**